### PR TITLE
Throw instead of silently failing with history v3

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

**Comment by taion** [comment]
Curses, why did I release history v2.5.0 before writing this?

**Comment by taion** [comment]
There are more duck-type-y ways to make this check, like failing in the transition manager if `history.listen` doesn't synchronously call back with an initial position, but this is the easiest and most straightforward way, and as far as I can tell it works.

**Comment by RobertWarrenGilmore** [comment]
Does history 2.5.0 cause the same bug as 3.x?

**Comment by taion** [comment]
There is no history 2.5.0.

**Comment by timdorr** [comment]
Push a 2.5.1 :)

File: modules/Router.js [hunk 1]
```diff
@@ -91,6 +98,13 @@ const Router = React.createClass({
     let { history } = this.props
     const { routes, children } = this.props
 
+    invariant(
+      !isUnsupportedHistory(history),
+      'You have provided a history object from created with history v3.x. ' +
```
Comment (Aweary) [inline modules/Router.js]: "from created with", is that a typo?

Comment (timdorr) [inline modules/Router.js]: Good catch!

**Comment by taion** [comment]
I'll release this tomorrow... about to call it a night and probably should be around in case this breaks people.

File: modules/Router.js (hunk 1 above)
Comment (taion) [inline modules/Router.js]: ):

**Pull request merged**